src/java.base/share/classes/java/util/ImmutableCollections.java

```diff
@@ -890,6 +890,39 @@ public <T> T[] toArray(T[] a) {
         }
     }
 
+    /**
+     * Creates a new Set from an untrusted array and checking for and
+     * rejecting null elements.
+     *
+     * @param <E> the Set's element type
+     * @param input the input array
+     * @return the new set
+     */
+    @SafeVarargs
+    @SuppressWarnings({"varargs", "unchecked", "rawtypes"})
+    static <E> Set<E> setFromArray(E... input) {
+        if (input.length == 0) {
```

Comment:
This special case can be removed, as no caller calls with an array with length <= 2